code/controllers/subsystems/timer.dm

```diff
@@ -259,6 +259,12 @@ SUBSYSTEM_DEF(timer)
 	// Add all timed events from the secondary queue as well
 	alltimers += second_queue
 
+	for (var/datum/timedevent/t in alltimers)
+		t.bucket_joined = FALSE
+		t.bucket_pos = -1
+		t.prev = null
+		t.next = null
```

Comment:
Don't really get why we do this here; unless I'm misreading the logic all of this is set again below except on deleted or irrelevant timers.